hypha/apply/activity/models.py

```diff
@@ -189,38 +213,93 @@ def __str__(self):
         return '{}: for "{}"'.format(self.get_type_display(), self.source)
 
     @classmethod
-    def visibility_for(cls, user):
+    def visibility_for(cls, user) -> List[str]:
+        """Gets activity visibility for a specified user
+
+        Args:
+            user:
+                [`User`][hypha.apply.users.models.User] to get visibility for
+
+        Returns:
+            A list of visibility strings
+        """
         if user.is_apply_staff:
             return [TEAM, APPLICANT, REVIEWER, PARTNER, ALL]
         if user.is_reviewer:
             return [REVIEWER, ALL]
         if user.is_finance or user.is_contracting:
             # for project part
             return [TEAM, APPLICANT, REVIEWER, PARTNER, ALL]
-        if user.is_applicant or user.is_partner:
-            return [
-                APPLICANT,
-                PARTNER,
-                ALL,
-            ]  # using partner just for existing activities.
+        if user.is_applicant:
+            return [APPLICANT, ALL, APPLICANT_PARTNERS]
+        if user.is_partner:
+            return [PARTNER, ALL, APPLICANT_PARTNERS]
 
         return [ALL]
 
     @classmethod
-    def visibility_choices_for(cls, user):
-        if user.is_applicant or user.is_partner:
+    def visibility_choices_for(
+        cls, user, submission_has_partner: Optional[bool] = False
+    ) -> List[Tuple[str, str]]:
+        """Gets activity visibility choices for the specified user
+
+        Args:
+            user:
+                The [`User`][hypha.apply.users.models.User] being given
+                visibility choices
+            submission_has_partner:
+                An optional boolean that indicates if partner should be
+                included in visibility options
+
+        Returns:
+            A list of tuples in the format of:
+
+            [(<visibility string>, <visibility display string>), ...]
+        """
+        if submission_has_partner:
+            if user.is_applicant:
+                return [
+                    (APPLICANT_PARTNERS, VISIBILITY[PARTNER]),
+                    (APPLICANT, VISIBILITY[TEAM]),
```

Comment:
For the applicant, it didn't make much sense to me to set the choice to `Applicant`. I figured this would remove the ambiguity.